provider/bigquery.go

```diff
@@ -335,17 +342,28 @@ func (q defaultBQQueries) createValuePlaceholderString(columns []TableColumn) st
 	return strings.Join(placeholders, ", ")
 }
 
-func (q defaultBQQueries) newBQOfflineTable(name string, columnType string) string {
+func (q defaultBQQueries) newBQOfflineTableQuery(name string, columnType string) string {
 	return fmt.Sprintf("CREATE TABLE `%s` (entity STRING, value %s, ts TIMESTAMP, insert_ts TIMESTAMP)", q.getTableName(name), columnType)
 }
 
-func (q defaultBQQueries) materializationCreate(tableName string, resultName string) string {
-	query := fmt.Sprintf(
-		"CREATE TABLE `%s` AS (SELECT entity, value, ts, row_number() over(ORDER BY entity) as row_number FROM "+
-			"(SELECT entity, ts, value, row_number() OVER (PARTITION BY entity ORDER BY ts DESC, insert_ts DESC) "+
-			"AS rn FROM `%s`) t WHERE rn=1)", q.getTableName(tableName), q.getTableName(resultName))
+func (q defaultBQQueries) materializationCreate(tableName string, schema ResourceSchema, resourceLocation pl.SQLLocation) string {
+	var sb strings.Builder
+	sb.WriteString(fmt.Sprintf("CREATE OR REPLACE TABLE `%s` AS ", tableName))
```

Comment:
Why is this a table and not a view? Not saying it should be a view just needed to understand, also explain what htis does